File: src/main/java/com/web/pages/LoginPage.java
package com.web.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class LoginPage extends BaseWebPage {

    // --- Locators (Private to encapuslate them) ---
    private By usernameInput = By.id("username");
    private By passwordInput = By.id("password");
    // The "Continue" button is used for both email and password submission
    private By actionButton = By.name("action");

    // --- Constructor ---
    public LoginPage(WebDriver driver) {
        super(driver);
    }

    // --- Page Actions ---

    public void enterUsername(String username) {
        writeText(usernameInput, username);
    }

    public void enterPassword(String password) {
        writeText(passwordInput, password);
    }

    public void clickContinue() {
        click(actionButton);
    }

    /**
     * Convenience method to perform the full login flow.
     * This matches the logic: Enter User -> Click -> Enter Pass -> Click
     */
    public void performLogin(String username, String password) {
        enterUsername(username);
        clickContinue();
        enterPassword(password);
        clickContinue();
    }
}

File: pages/web/LoginPage.ts
import { Page, Locator } from "@playwright/test";
import BaseWebPage from "./BaseWebPage";

class LoginPage extends BaseWebPage {
    private usernameInput: Locator;
    private passwordInput: Locator;
    private actionButton: Locator;

    constructor(page: Page) {
        super(page);
        this.usernameInput = page.locator("#username");
        this.passwordInput = page.locator("#password");
        this.actionButton = page.locator('[name="action"]');
    }

    async enterUsername(username: string): Promise<void> {
        await this.writeText(this.usernameInput, username);
    }

    async enterPassword(password: string): Promise<void> {
        await this.writeText(this.passwordInput, password);
    }

    async clickContinue(): Promise<void> {
        await this.click(this.actionButton);
    }

    async performLogin(username: string, password: string): Promise<void> {
        await this.enterUsername(username);
        await this.clickContinue();
        await this.enterPassword(password);
        await this.clickContinue();
    }
}

export default LoginPage;
